src/infrastructure/persistence/PsqlRoleRepository.ts

```diff
@@ -0,0 +1,113 @@
+import { RoleRepository } from '../../domain/model/authentication/RoleRepository';
+import knex from 'knex';
+import { Role } from '../../domain/model/authentication/Role';
+import { PermissionAssignmentAction } from '../../domain/model/authentication/PermissionAssignmentAction';
+import { Permission } from '../../domain/model/authentication/Permission';
+import { AssignmentId } from '../../domain/model/authentication/AssignmentAction';
+import { AssignmentActionType } from '../../domain/model/authentication/AssignmentActionType';
+import { UserId } from '../../domain/model/user/UserId';
+
+export class PsqlRoleRepository implements RoleRepository {
+  constructor(private db: knex) {}
+
+  async findByName(name: string): Promise<Role | null> {
+    const roleRows: any[] = await this.db('role')
+      .select([
+        'role.name as role_name',
```

Comment:
Could do `as roleName` here as well, just to stick to JS conventions, and follow the patterns we're using elsewhere.